### PR TITLE
Bug 1261849 - Hello conversation opens while Firefox is in never reme…

### PR DESCRIPTION
**Pull request by mancas**
…mber history mode

### CODE REVIEW

File: add-on/chrome/bootstrap.js [hunk 1]
```diff
@@ -211,6 +211,11 @@ var WindowListener = {
             });
           };
 
+          if (PrivateBrowsingUtils.isWindowPrivate(window)) {
+            resolve(null);
+            return;
+          }
```
Comment (Mardak) [inline add-on/chrome/bootstrap.js]: This doesn't need to be inside the `Promise` callback. You can check immediately inside `openPanel` and `return Promise.resolve()` to short circuit.

Comment (Mardak) [inline add-on/chrome/bootstrap.js]: I'm also not sure if we should `reject()` instead as the UITour code will call an optional `aOpenCallback` on success: https://dxr.mozilla.org/mozilla-central/source/browser/components/uitour/UITour.jsm#1720-1742

Comment (Mardak) [inline add-on/chrome/bootstrap.js]: Although, it's quite possible the only uses of the `Promise` is for testing, e.g., https://dxr.mozilla.org/mozilla-central/source/browser/components/uitour/test/head.js#167-174 to indicate the test should continue. We should have a mochitest similar to https://dxr.mozilla.org/mozilla-central/source/browser/components/uitour/test/browser_showMenu_controlCenter.js to make sure the panel opens in non-private browsing and doesn't in private browsing.

File: add-on/chrome/test/mochitest/browser_panel_privateBrowsing.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+add_task(function* test_init_copy_panel_private() {
+  let win = OpenBrowserWindow({ private: true });
+  yield new Promise(resolve => win.addEventListener("load", resolve));
+  yield win.LoopUI.openPanel();
+
+  Assert.equal(win.document.getElementById("loop-notification-panel"), null, "Panel doesn't exist for private browsing");
+
+  win.close();
+});
```
Comment (mancas) [inline add-on/chrome/test/mochitest/browser_panel_privateBrowsing.js:37]: @Mardak , could you please review this tests? For some reason I'm getting an error when running them:

Error: Found an unexpected browser window at the end of test run

Comment (Standard8) [inline add-on/chrome/test/mochitest/browser_panel_privateBrowsing.js:37]: You might want to switch to:

``` js
let win = yield BrowserTestUtils.openNewBrowserWindow();
yield win.LoopUI.openPanel();
```

BrowserTestUtils has code for waiting for load etc.

You probably also want to change the `win.close()` to a cleanup function at the start of the test - this ensures it gets called even if the test fails, so you're less likely to impact other tests if something goes wrong (and the failures are clearer).

``` js
  registerCleanupFunction(function* () {
    if (win) {
      yield BrowserTestUtils.close(win);
    }
  });
});
```

File: add-on/chrome/test/mochitest/browser_panel_privateBrowsing.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  yield win.LoopUI.openPanel();
+
+  Assert.equal(win.document.getElementById("loop-notification-panel"), null, "Panel doesn't exist for private browsing");
+});
+
+registerCleanupFunction(function* () {
+  if (win) {
+    yield BrowserTestUtils.closeWindow(win);
+  }
+});
```
Comment (mancas) [inline add-on/chrome/test/mochitest/browser_panel_privateBrowsing.js:37]: @Standard8 I don't really know what I'm doing wrong. Both tests seems to work but when I run them, an extra window is being opening. Any ideas?

**Pull request merged**